### PR TITLE
Improve licence enforcer coverage

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

File: tests/scripts/enforce_licence.py [hunk 1]
```diff
@@ -28,24 +28,51 @@
...
   'src/main.c'
 ]
 
 IGNORED_DIRECTORIES = [
-    'Bootloader/firmware/src/system_config',
-    'firmware/src/system_config',
-    'src/system_config',
-    'tests/src/system_config',
-    'tests/boot_src/system_config',
+    'Bootloader/firmware/src/system_config/ethernet_sk2/framework/',
```
Comment (nomis52) [inline tests/scripts/enforce_licence.py]: Change to Bootloader/firmware/src/system_config & firmware/src/system_config

File: tests/scripts/enforce_licence.py [hunk 2]
```diff
@@ -28,24 +28,51 @@
...
+  'firmware/src/system_config/ethernet_sk2/system_init.c',
+  'firmware/src/system_config/ethernet_sk2/system_interrupt.c',
+  'firmware/src/system_config/ethernet_sk2/system_pipeline.h',
+  'firmware/src/system_config/ethernet_sk2/system_tasks.c',
+  'firmware/src/system_config/number8/system_config.h',
+  'firmware/src/system_config/number8/system_definitions.h',
+  'firmware/src/system_config/number8/system_init.c',
+  'firmware/src/system_config/number8/system_interrupt.c',
+  'firmware/src/system_config/number8/system_pipeline.h',
+  'firmware/src/system_config/number8/system_tasks.c',
```
Comment (nomis52) [inline tests/scripts/enforce_licence.py]: Do we need these if we've added the directories below? I hope not.

File: tests/scripts/enforce_licence.py (hunk 1 above)
Comment (peternewman) [inline tests/scripts/enforce_licence.py]: That's what you had, but there are valid licensed files in there that should be checked. I've just explicitly blacklisted the autogenerated stuff.

Comment (nomis52) [inline tests/scripts/enforce_licence.py]: There shouldn't be any now, those directories are all autogenerated,

Comment (peternewman) [inline tests/scripts/enforce_licence.py]: Ah yes, I was out of step, it's all in boardcfg now isn't it?

Comment (nomis52) [inline tests/scripts/enforce_licence.py]: Yup

Comment (peternewman) [inline tests/scripts/enforce_licence.py]: Fixed

File: tests/scripts/enforce_licence.py (hunk 2 above)
Comment (peternewman) [inline tests/scripts/enforce_licence.py]: Removed.

**Comment by nomis52** [comment]
Can you move the scripts directory to the top level as part of this as well. I've done everything else.

**Comment by peternewman** [comment]
Moved the file.

**Comment by peternewman** [comment]
Just thinking @nomis52 given we need to somehow handle the path depth for licence header comments in boardconfig, shall I extend the enforcer to specify a folder and depth, and it will then complain if you don't have the right depth for the name within that folder?

**Comment by nomis52** [comment]
It may be easier just to change the file paths to remove the directories :)

**Comment by peternewman** [comment]
You mean within the file headers?

**Comment by nomis52** [comment]
Yeah

change

number8/board_init.h to just board_init.h

**Comment by peternewman** [comment]
I thought you wanted the folder in place for clarity? I don't think it should be too hard to code up if you want to keep it.

**Comment by nomis52** [comment]
It's not worth keeping it it causes trouble.

**Comment by peternewman** [comment]
It was easy enough to fix within the licence enforcer script, so that's now done (and a few bits it caught fixed). Merge if you're happy @nomis52

File: scripts/enforce_licence.py [hunk 3]
```diff
@@ -249,7 +255,20 @@ def CheckLicenceForFile(file_name, licence, lang, diff, fix):
...
-    expected_line = TransformLine(os.path.basename(file_name), lang)
+    relative_path = os.path.relpath(os.path.dirname(file_name), os.getcwd())
+    expected_filename_depth = 0
+    for dir_name, depth in FILENAME_DEPTH_OVERRIDE.iteritems():
+      if relative_path.startswith(dir_name):
+        expected_filename_depth = depth
+    expected_filename_parts = [os.path.basename(file_name)]
+    depth_path = relative_path
+    for i in range(expected_filename_depth):
+      (depth_path, folder) = os.path.split(depth_path)
```
Comment (nomis52) [inline scripts/enforce_licence.py]: No need for parens here

**Comment by nomis52** [comment]
Just a single comment. Merge when fixed.

File: scripts/enforce_licence.py (hunk 3 above)
Comment (peternewman) [inline scripts/enforce_licence.py]: Done.

**Pull request merged**